internal/pkg/translator/obj2cil.go

```diff
@@ -0,0 +1,142 @@
+)
+
+func Object2CIL(cli client.Client, sp *selxv1alpha2.SelinuxProfile) (string, error) {
+	cilbuilder := strings.Builder{}
+	cilbuilder.WriteString(getCILStart(sp))
+	for _, inherit := range sp.Spec.Inherit {
+		usage, err := getInheritName(cli, inherit, sp.GetNamespace())
+		if err != nil {
+			return "", err
+		}
+		cilbuilder.WriteString(getCILInheritline(usage))
+	}
+	for ttype, tclassMap := range sp.Spec.Allow {
+		for tclass, perms := range tclassMap {
+			cilbuilder.WriteString(getCILAllowLine(sp, ttype, tclass, perms))
+		}
+	}
+	cilbuilder.WriteString(getCILEnd(sp))
+	return cilbuilder.String(), nil
+}
+
+func getCILStart(sp *selxv1alpha2.SelinuxProfile) string {
+	return fmt.Sprintf("(block %s_%s\n", sp.GetName(), sp.GetNamespace())
+}
+
+func getInheritName(
+	cli client.Client,
+	ancestorRef selxv1alpha2.PolicyRef,
+	namespace string,
+) (string, error) {
+	switch ancestorRef.Kind {
+	// We default to System if Kind is left empty
+	case selxv1alpha2.SystemPolicyKind, "":
+		return handleInheritSystemPolicy(cli, ancestorRef)
+	case "SelinuxPolicy", "RawSelinuxPolicy":
+		return handleInheritSPOPolicy(cli, ancestorRef, namespace)
+	}
+	return "", fmt.Errorf("unknown inherit kind for entry: %s/%s", ancestorRef.Kind, ancestorRef.Name)
+}
+
+func handleInheritSPOPolicy(
+	cli client.Client,
+	ancestorRef selxv1alpha2.PolicyRef,
+	namespace string,
+) (string, error) {
+
+	var ancestor selxv1alpha2.SelinuxProfileObject
+	kind := ancestorRef.Kind
+	key := types.NamespacedName{Name: ancestorRef.Name, Namespace: namespace}
+	switch ancestorRef.Kind {
+	// We default to SelinuxPolicy if Kind is left empty
+	case "SelinuxPolicy":
+		kind = "SelinuxPolicy"
+		ancestor = &selxv1alpha2.SelinuxProfile{}
+	case "RawSelinuxPolicy":
+		ancestor = &selxv1alpha2.RawSelinuxProfile{}
+	default:
+		return "", fmt.Errorf("Unknown inherit kind for entry: %s/%s", ancestorRef.Kind, ancestorRef.Name)
+	}
+	err := cli.Get(context.Background(), key, ancestor)
+	if err != nil && errors.IsNotFound(err) {
+		message := fmt.Sprintf("couldn't find inherit reference %s/%s", kind, ancestorRef.Name)
+		return "", fmt.Errorf("%s: %w", message, err)
+	}
+
+	// TODO(jaosorior): Handle dependencies... we could force a waiting period
+	// until the ancestor policy would be installed
+	return ancestor.GetPolicyUsage(), nil
+}
+
+func handleInheritSystemPolicy(
+	cli client.Client,
+	ancestorRef selxv1alpha2.PolicyRef,
+) (string, error) {
+	spod, err := common.GetSPOD(context.Background(), cli)
+	if err != nil {
+		return "", fmt.Errorf("couldn't get spod to verify system inheritance: %w", err)
+	}
+
+	for idx := range spod.Spec.SelinuxOpts.AllowedSystemProfiles {
+		prof := spod.Spec.SelinuxOpts.AllowedSystemProfiles[idx]
+		if prof == ancestorRef.Name {
+			return prof, nil
+		}
+	}
+	msg := fmt.Sprintf("system profile %s not in SecurityProfilesOperatorDaemon's allow list", ancestorRef.Name)
+	return "", fmt.Errorf("%s: %w", msg, err)
```

Comment:
why not wrapf?

Reply:
uuuh... what's the difference between `Errorf` with `%w` and `Wrapf` ?

Reply:
Functionally, I think none, except maybe unwrapping?. I thought that Wrap was the preferred way these days and I seem to remember that one of the linters complained when I did not use Wrap.
It's really not the biggest deal.